src/main/java/org/jenkinsci/plugins/workflow/log/FileLogStorage.java

```diff
@@ -126,17 +126,28 @@ private void checkId(String id) throws IOException {
         assert Thread.holdsLock(this);
         if (!Objects.equals(id, lastId)) {
             bos.flush();
-            long pos = os.getChannel().position();
-            if (id == null) {
-                indexOs.write(pos + "\n");
-            } else {
-                indexOs.write(pos + " " + id + "\n");
+            // FileChannel methods such as position() close the channel (and the stream it was derived from) if the
+            // current thread is interrupted. We must not allow interruption on some arbitrary thread attempting to
+            // write to the log to close the stream irrevocably, so we clear the interrupt status before using the
+            // channel and then restore the interrupt status afterwards.
+            boolean interrupted = Thread.interrupted();
+            try {
+                long pos = os.getChannel().position();
```

Comment:
It would suffice to have the `finally` block run as soon as `Channel.position` has been called, right? I.e., the operations on `indexOs` need not be included?